### PR TITLE
Update `box_shadow:` to use functional classes

### PR DESCRIPTION
**Pull request by manuelpuyol**
according to https://primer-css-git-mkt-color-modes-docs-primer.vercel.app/css/support/v16-migration#shadow the mapping is 1:1, so we can simply change the classes

### CODE REVIEW

File: app/lib/primer/classify.rb [hunk 1]
```diff
@@ -252,9 +252,11 @@ def extract_value(memo, key, val, breakpoint)
           memo[:classes] << "#{key.to_s.dasherize}#{breakpoint}-n#{val.abs}"
         elsif key == BOX_SHADOW_KEY
           memo[:classes] << if val == true
-                              "box-shadow"
+                              "color-shadow-small"
+                            elsif val == :none
```
Comment (srt32) [inline app/lib/primer/classify.rb:256]: Should we check for `val == false` too?

File: app/lib/primer/classify.rb [hunk 2]
```diff
@@ -252,9 +252,11 @@ def extract_value(memo, key, val, breakpoint)
...
+                              "box-shadow-none"
                             else
-                              "box-shadow-#{val.to_s.dasherize}"
+                              "color-shadow-#{val.to_s.dasherize}"
```
Comment (srt32) [inline app/lib/primer/classify.rb:259]: I would have expected some spec to change or fail. Is that something we could add?

Comment (manuelpuyol) [reply app/lib/primer/classify.rb:259]: it probably will, I totally forgot about the spec 😅

File: app/lib/primer/classify.rb (hunk 1 above)
Comment (manuelpuyol) [reply app/lib/primer/classify.rb:256]: hmm we never used it, but don't see why not

File: test/lib/classify_test.rb [hunk 3]
```diff
@@ -322,11 +322,13 @@ def test_font_weight
   end
 
   def test_box_shadow
-    assert_generated_class("box-shadow",             { box_shadow: true })
-    assert_generated_class("box-shadow-medium",      { box_shadow: :medium })
-    assert_generated_class("box-shadow-large",       { box_shadow: :large })
-    assert_generated_class("box-shadow-extra-large", { box_shadow: :extra_large })
-    assert_generated_class("box-shadow-none",        { box_shadow: :none })
+    assert_generated_class("color-shadow-small",       { box_shadow: true })
+    assert_generated_class("color-shadow-small",       { box_shadow: true })
```
Comment (srt32) [inline test/lib/classify_test.rb:326]:
```suggestion
    assert_generated_class("color-shadow-small",       { box_shadow: true })
```

dup line?

**Review by srt32 — APPROVED** [review]

**Pull request merged**